_hardware/cypherockx1.md

```diff
@@ -16,52 +16,138 @@ providerWebsite: https://cypherock.com/
 
-> - The code for the X1 wallet **will** be open source
-> - Each shard will be stored in a EAL6+ tamper resistant secure element chips. As comparison, an iPhone has the same level of security chip for securing fingerprints.
-> - Nothing will be stored in the X1 Wallet on which the computation will be done.
-> - Keys can be recovered by tapping any 2 out of the 4 cyCards on the X1 Wallet. Each cyCard can also be protected by a PIN just like any other hardware wallet. The recovered keys are stored in a volatile memory on the X1 Wallet which gets wiped out as soon as the operation is done.
-> - The user can store 3 wallet recovery phrases in a single product.
-> - The cyCards can be distributed around different places to increase security of the funds and protect against $5 wrench attacks.
+### Alternatively, one can follow below commands in a *powershell* window to build the Cypherock X1 Wallet firmware:
+* For Initial firmware - unsigned binary will be generated at build/initial/Cypherock-Initial.bin
+        
+        docker build . --file Dockerfile --tag x1-wallet-app-env
+        mkdir -p build
+        docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash
+        mkdir -p /dist/initial && cd /home
+        cd build/Release/
+        cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Initial -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        ninja
+        cp Cypherock-*.* /dist/initial && cp -a /dist/. /out
+        exit
 
-From the [FAQ](https://www.cypherock.com/faq/): Can the company somehow steal the digital assets of the user?
+* For Main firmware - unsigned binary will be generated at build/main/Cypherock-Main.bin
+        
+        docker build . --file Dockerfile --tag x1-wallet-app-env
+        mkdir -p build
+        docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash
+        mkdir -p /dist/main && cd /home
+        cd build/Release/
+        cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Main -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        ninja
+        cp Cypherock-*.* /dist/main && cp -a /dist/. /out
+        exit
+
+
+### Reproducing the Cypherock X1 Wallet Initial firmware v0.4.773 on Windows 11
+* On Git Bash Shell, checkout using tag:
+        
+        MINGW64 ~/Desktop/Cypherock/X1_wallet/Firmware/v0.4.773 ((v0.4.773)) $ git clone git@github.com:Cypherock/x1_wallet_firmware.git .
+        MINGW64 ~/Desktop/Cypherock/X1_wallet/Firmware/v0.4.773 ((v0.4.773)) $ git checkout v0.4.773
+        .
+        .
+
+* On Powershell, setup Docker and compile:
+
+        PS C:\Users\LENEVO\Desktop\Cypherock\X1_wallet\Firmware\v0.4.773>  docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash docker build . --file Dockerfile --tag x1-wallet-app-env
```

Comment:
This is two docker commands in one line, probably in reverse order.

Reply:
I have fixed the order.